src/core/sys/posix/sys/types.d

```diff
@@ -604,6 +637,37 @@ version (CRuntime_Glibc)
 
     alias c_ulong pthread_t;
 }
+else version( CRuntime_Musl )
+{
+    union pthread_attr_t
```

Comment:
Again, please separate the definitions per architecture. For reference:
http://git.musl-libc.org/cgit/musl/tree/arch/i386/bits/alltypes.h.in
http://git.musl-libc.org/cgit/musl/tree/arch/x86_64/bits/alltypes.h.in
http://git.musl-libc.org/cgit/musl/tree/arch/arm/bits/alltypes.h.in
http://git.musl-libc.org/cgit/musl/tree/arch/aarch64/bits/alltypes.h.in
http://git.musl-libc.org/cgit/musl/tree/arch/mipsn32/bits/alltypes.h.in
http://git.musl-libc.org/cgit/musl/tree/arch/mips64/bits/alltypes.h.in
...